### PR TITLE
Enable UEFI for QEMU VMs

### PR DESCRIPTION
**Pull request by temerkhanov**
Enable running QEMU VMs with UEFI. This apparently fixes Supermicro.

### CODE REVIEW

File: pkg/pillar/hypervisor/kvm.go [hunk 1]
```diff
@@ -102,6 +102,19 @@ const qemuConfTemplate = `# This file is automatically generated by domainmgr
 [realtime]
   mlock = "off"
 
+[drive]
```
Comment (rvs) [inline pkg/pillar/hypervisor/kvm.go:105]: Wait, why are we changing this all of a sudden? What's wrong with the `ovmf.bin` that has worked for us for quite some time? @temerkhanov

File: pkg/pillar/hypervisor/kvm.go [hunk 2]
```diff
@@ -452,27 +486,7 @@ func (ctx kvmContext) CreateDomConfig(domainName string, config types.DomainConf
 		PCIId, DiskID, SATAId int
 		AioType               string
 		types.DiskStatus
-	}{Machine: ctx.devicemodel, PCIId: 4, DiskID: 0, SATAId: 0, AioType: "threads"}
-
-	var osver []string
-	var major, minor, patch uint64
-
-	osver = strings.SplitN(getOsVersion(), ".", 3)
```
Comment (rvs) [inline pkg/pillar/hypervisor/kvm.go:460]: We can discuss separately whether dropping this is a good idea -- but please lets make it part of a separate PR @temerkhanov

File: pkg/pillar/hypervisor/qmp.go [hunk 3]
```diff
@@ -15,12 +15,24 @@ import (
 const sockTimeout = 10 * time.Second
 
 func execRawCmd(socket, cmd string) ([]byte, error) {
-	monitor, err := qmp.NewSocketMonitor("unix", socket, sockTimeout)
+	var retry int = 3
```
Comment (rvs) [inline pkg/pillar/hypervisor/qmp.go:18]: Must say -- I kind of agree with revive here @temerkhanov

File: pkg/pillar/hypervisor/qmp.go [hunk 4]
```diff
@@ -15,12 +15,24 @@ import (
...
+	logrus.Infof("executing QMP command: %s", cmd)
+	var err error
+	var monitor *qmp.SocketMonitor
+
+	for retry >= 0 {
```
Comment (rvs) [inline pkg/pillar/hypervisor/qmp.go:23]: Can you please elaborate why is this necessary?

File: pkg/xen-tools/patches-4.14.0/13-qemu-Revamp-mmap-allocation-routine.patch [hunk 5]
```diff
@@ -0,0 +1,124 @@
+From 3ad5ce46a00385fc59adb5ba9f7c4b358e346bbf Mon Sep 17 00:00:00 2001
```
Comment (rvs) [inline pkg/xen-tools/patches-4.14.0/13-qemu-Revamp-mmap-allocation-routine.patch:1]: What is exact root cause that requires us to have this patch?

**Review by rvs — CHANGES_REQUESTED** [review]
There's a LOT of questions I have about this one @temerkhanov and it obviously will require a lot of testing before we can a release with it.

Also, I suggest that as you combine it with #1859 so that we can take care of all UEFI issues in one PR (and thus minimize testing impact as well)

**Review by eriknordmark — CHANGES_REQUESTED** [review]
@temerkhanov The diffs seem to have changes from unrelated PR which have gone into master recently, plus your PR has lots of commits.
So I can't tell what you are proposing to change.

FWIW what works for me is to 
 git remote add upstream https://github.com/lf-edge/eve
 git fetch upstream master
 git rebase -i upstream/master
and as part of that rebase squash your commits (typically into a single one unless you have multiple unrelated changed in your PR; those can be separate commits)
Hence never do a git pull into your bramch because you end up with merge commits which will present us from pulling to master and maintain a linear history

File: pkg/pillar/hypervisor/kvm.go (hunk 1 above)
Comment (temerkhanov) [reply pkg/pillar/hypervisor/kvm.go:105]: ovmf.bin apparently was hanging on startup on Supermicro HW that I have. This PR adds an ability to build a debug version and see what is going on in such occasions

File: pkg/pillar/hypervisor/kvm.go (hunk 2 above)
Comment (temerkhanov) [reply pkg/pillar/hypervisor/kvm.go:460]: Moved to another PR

File: pkg/pillar/hypervisor/qmp.go (hunk 4 above)
Comment (temerkhanov) [reply pkg/pillar/hypervisor/qmp.go:23]: I saw QMP connection failures, supposedly due to a late qemu start

File: pkg/xen-tools/patches-4.14.0/13-qemu-Revamp-mmap-allocation-routine.patch (hunk 5 above)
Comment (temerkhanov) [reply pkg/xen-tools/patches-4.14.0/13-qemu-Revamp-mmap-allocation-routine.patch:1]: This one is merged from a separate PR, should reduce noise in the memory profilers at least.

File: pkg/pillar/hypervisor/kvm.go (hunk 2 above)
Comment (temerkhanov) [reply pkg/pillar/hypervisor/kvm.go]: Moved to a different PR

File: pkg/pillar/hypervisor/qmp.go (hunk 4 above)
Comment (temerkhanov) [reply pkg/pillar/hypervisor/qmp.go:23]: Move to a different PR

**Comment by rouming** [comment]
@eriknordmark can we please close this? not sure who picks this up for further changes and the actual profit.

**Comment by eriknordmark** [comment]
I don't see this PR making any progress.
